### DISCUSSION TITLE
翻訳記事のメタデータ（front-matter）に `short-title` を必須とすべきか

### DISCUSSION THREAD

**Discussion by debiru** [mozilla-japan/translation#699] [General]
#698 に関連して、以下のトピックを見つけました。

- 2023年4月25日のコメント https://github.com/mdn/yari/issues/8647#issuecomment-1520722736

> `short-title` の考え方は、フルタイトルの使用が不適切な文脈で使用できる代替タイトルであるということです。一般的には、`short-title` が表示される文脈（サイドバーやパンくずのようなもの）が理由です。
>
> `short-title` は、`title` と同様にローカライズ可能なメタデータの一部として扱い、en-USから継承しない方が無難だと思います。Web/APIでは、short-titleの値は常に識別子であり、ローカライズ可能ではありませんが、常にそうであるという保証はありませんし、実際、少なくとも1つのケースではそうではありません： https://github.com/mdn/content/blob/23346b58b7ef97cc21f34cb38d23b23487a9d097/files/en-us/web/progressive_web_apps/reference/index.md?plain=1#L2-L3.

- 2023年4月25日のコメント https://github.com/mdn/translated-content/issues/7412#issuecomment-1520518105

> MDNチームの他のメンバーと話したところ、`short_title` もローカライズ可能なので、それも翻訳コンテンツとして残しておくべきだと指摘されました！

メタデータについて、`title` と `slug` と `l10n` と `original_slug` 以外は削除する方針にすべきという話があったかと思いますが、その後少し方針（記述内容）の変更があったようです。

翻訳側の記事のメタデータには、

- `original_slug` は残さなくてよい
- `short_title` は残す

と更新されています。

`original_slug` の「残さなくてよい」が「削除すべき」という意味のなるのかどうかは分かりませんが、`original_slug` については記述していても悪影響があるわけではないので、現時点では削除しなくてもよいかなと思っています。（すでに設定されている記事について、それが不要であると判断できるなら削除していく方針でもよいとは思います。）

`short_title` については、その用途から、翻訳されるべきと言及されています。これについて、日本語コミュニティでは `short_title` の記述の必要性についてこれまで情報共有されていませんでしたが、記述の必要性について一考する必要があるように思いました。

https://github.com/mdn/content/blob/23346b58b7ef97cc21f34cb38d23b23487a9d097/files/en-us/web/progressive_web_apps/reference/index.md?plain=1#L2-L3. の例を見ると、`short-title` には、その文書のカテゴリを暗黙的に継承するような前提で `short-title` を略記できるようです。つまり、何の「リファレンス」かを示さずとも「リファレンス」という `short-title` をつけることが適していると判断できるケースがあるようです。

`short-title` が表示される箇所や、その設定の方針（略記の仕方）について理解を深めた上で、日本語コミュニティ内で `short-title` の設定を行っていくようにすべきかと思ったのですがいかがでしょうか。なお、韓国語コミュニティではすでにこれを必須とする方針変更があったようです。 https://github.com/mdn/translated-content/pull/13068

**Comment by potappo** [comment]
翻訳して問題が出ないか（この場合、見た目的なものかと思います）などの確認は必要かと思いますが、short-title を日本語版でも翻訳するという方針について、問題ないと思います。

>short-title が表示される箇所や、その設定の方針（略記の仕方）について

上記情報が十分に集まった時点で、翻訳ガイドなどに追記するという方針で良いかと思います。

**Reply by debiru** [reply]
> 上記情報が十分に集まった時点で、翻訳ガイドなどに追記するという方針で良いかと思います。

ありがとうございます。そうですね。急ぐものでもないので、適当なタイミングで手順等を追加できればよさそうですね。

なんとなく、現実的にはページの編集ごとに `short-title` を設定するというよりは、サイドバー等の見栄えの調整という名目で、カテゴリ毎にまとめて複数ページに対して `short-title` を設定するというタスクとして処理した方が効率が良さそうに思えました。
